targets/TARGET_ONSEMI/TARGET_NCS36510/clock.h

```diff
@@ -83,6 +83,7 @@
 #define CLOCK_ENABLE(a)		CLOCKREG->PDIS.WORD &= ~(1 << a)
 #define CLOCK_DISABLE(a)	CLOCKREG->PDIS.WORD |= (uint32_t)(1 << a)
 
+#define CLOCK_IS_ENABLED(a)     ((CLOCKREG->PDIS.WORD >> a) & 1)?0:1
```

Comment:
This is dangerous due to lack of enclosing brackets. Indeed, I believe all your uses of it actually fail.